src/app/users/user-profile.component.ts

```diff
@@ -116,21 +118,30 @@ export class UserProfileComponent implements OnInit {
     }
   }
 
-  onSubmit() {
+  saveUserDetails() {
+    this.savingInProgress = true;
     this.usersService.updateUser(this.buildUserRequestData()).subscribe(
       user => {
+        this.savingInProgress = false;
```

Comment:
There is no need to set `savingInProgress` flag to false twice. This can be achieved using `finally` extension method https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/operators/finally.md
https://stackoverflow.com/questions/44770999/observables-complete-vs-finally-vs-done